lib/streams/intersperse-stream.js

```diff
@@ -0,0 +1,46 @@
+var Transform = require('../transform');
+var inherits = require('util').inherits;
+
+/**
+ * IntersperseStream intersperses a seperator between chunks in the stream
+ *
+ * @class IntersperseStream
+ * @constructor
+ * @extends Transform
+ * @param {String|Object|Number} [seperator='\n'] - Seperator to push between chunks
+ * @param {Object} [options] - Stream options object
+ */
+function IntersperseStream(seperator, options) {
+	if(typeof seperator === 'object') {
+		options = seperator;
+		seperator = null;
+	}
+	options = !options ? {} : options;
+
+	Transform.call(this, options);
+
+	this._buffer = null;
+	this._seperator = (seperator === null || seperator === undefined) ? '\n' : seperator;
+	this._first = true;
+}
+inherits(IntersperseStream, Transform);
+
+IntersperseStream.prototype._transform = function(chunk, encoding, cb) {
+	if(!this._buffer) {
+		this._buffer = chunk;
+		return cb();
+	}
+
+	this.push(this._buffer);
+	this.push(this._seperator);
```

Comment:
This will cause the stream to always end with a delimiter.  "intersperse" suggests that the value only be in between elements.  I'd suggest not making the stream end with a delimiter by default, and adding an option to use that behavior.